bundles/config/org.eclipse.smarthome.config.discovery.mdns/src/main/java/org/eclipse/smarthome/config/discovery/mdns/internal/MDNSDiscoveryService.java

```diff
@@ -114,18 +117,40 @@ protected void stopBackgroundDiscovery() {
 
     @Override
     protected void startScan() {
+        startScan(false);
+    }
+
+    private void startScan(boolean isBackground) {
         scheduler.schedule(new Runnable() {
             @Override
             public void run() {
-                scan();
+                scan(isBackground);
             }
         }, 0, TimeUnit.SECONDS);
     }
 
-    private void scan() {
+    /**
+     * Scan has 2 different behaviours. background/ foreground. Background scans can
+     * have much higher timeout. Foreground scans have only a short timeout as human
+     * users may become impatient. The underlying reason is that the jmDNS
+     * implementation {@code MDNSClient#list(String)} has a default timeout of 6
+     * seconds when no ServiceInfo is found. When there are many participants,
+     * waiting 6 seconds for each non-existent type is too long.
+     * 
+     * @param isBackground
+     *            true if it is background scan, false otherwise.
+     */
+    private void scan(boolean isBackground) {
         for (MDNSDiscoveryParticipant participant : participants) {
-            ServiceInfo[] services = mdnsClient.list(participant.getServiceType());
-            logger.debug("{} services found for {}", services.length, participant.getServiceType());
+            long start = System.currentTimeMillis();
+            ServiceInfo[] services;
+            if (isBackground) {
+                services = mdnsClient.list(participant.getServiceType());
+            } else {
+                services = mdnsClient.list(participant.getServiceType(), FOREGROUND_SCAN_TIMEOUT);
+            }
+            logger.debug("{} services found for {}; duration: {}ms", services.length, participant.getServiceType(),
+                    (System.currentTimeMillis() - start));
```

Comment:
What's the reason for the braces around `System.currentTimeMillis() - start`? Aren't they rather useless?